RustFxA/RustFirefoxAccounts.swift

```diff
@@ -95,13 +100,11 @@ open class RustFirefoxAccounts {
         return id
     }
 
-    public static func reconfig(_ completion: ((FxAccountManager) -> Void)? = nil) {
+    @discardableResult
+    public static func reconfig(prefs: Prefs) -> Deferred<FxAccountManager> {
+        isInitializingAccountManager = false
         shared.accountManager = Deferred<FxAccountManager>()
-        let manager = shared.createAccountManager()
-        manager.initialize() { _ in
-            shared.accountManager.fill(manager)
-            completion?(manager)
-        }
+        return startup(prefs: prefs)
```

Comment:
The old function was over-complicating a simple thing; just reset the `shared.accountManager` to a fresh `Deferred`, and then call `startup` as normal. Most callers of this are not making use of the result (it is a special-case function and not really part of normal flow except for the notification service), so `discardableResult` is appropriate here.

Notice that `isInitializingAccountManager` needs to be set to false in order for `shared.accountManager` to get re-initialized. I thought about making a tuple to tightly bond these together:
`sharedAccountManager = (manager: Deferred<...>, isInitializing: Bool)`
But I think that is overkill and would require making a bunch more code changes.